File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/java/field_assign_to_array_item_in_function2.java
class Foo {
    String field = null;
}
class A {
    void func(Foo fo111, String[] fo222)
    {
        fo222[1] = fo111.field;
    }

    int main()
    {
        Foo f11;
        String[] f22 = new String[10];
        func(f11, f22);
        return f22[1].length();  // should be detected here
    }
}

File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/field_assign_to_array_item_in_function2.cpp
class Foo {
public:
    int *field;
    int *other;
};

void func(Foo* fo111, int* fo222[10]) {
    fo222[1] = fo111->field;
}

int main() {
    Foo f11;
    int* f22[10];
    f11.field = 0;
    func(&f11, f22);
    return *f22[1]; // should be detected here
}
